pennylane/ops/functions/equal.py

```diff
@@ -302,6 +303,24 @@ def _equal_hamiltonian(op1: Hamiltonian, op2: Observable, **kwargs):
     return op1.compare(op2)
 
 
+@_equal.register
+def _equal_parametrized_evolution(op1: ParametrizedEvolution, op2: ParametrizedEvolution, **kwargs):
+    # check times match
+    if not qml.math.allclose(op1.t, op2.t):
+        return False
+
+    # check parameters passed to operator match
+    if not _equal_operators(op1, op2, **kwargs):
+        return False
+
+    # check H.coeffs match
+    if not all(c1 == c2 for c1, c2 in zip(op1.H.coeffs, op2.H.coeffs)):
+        return False
+
+    # check that all the base operators on op1.H and op2.H match
+    return all(equal(o1, o2, **kwargs) for o1, o2 in zip(op1.H.ops, op2.H.ops))
```

Comment:
Can't we use `_equal_operator` here as well? :thinking:

Reply:
No - this should dispatch to the specific relevant function. The `_equal_operator` function is a fallback for operators that don't have a more specific function to handle them. All the `SymbolicOp` and `CompositeOp` operators need to go to their own functions. 

What if one of the operators the ParametrizedHamiltonian is `SProd`, for example?